src/rust/cryptography-x509-validation/src/lib.rs

```diff
@@ -10,3 +10,297 @@ pub mod ops;
 pub mod policy;
 pub mod trust_store;
 pub mod types;
+
+use std::collections::HashSet;
+
+use crate::certificate::cert_is_self_issued;
+use crate::types::{DNSConstraint, IPAddress, IPConstraint};
+use crate::ApplyNameConstraintStatus::{Applied, Skipped};
+use cryptography_x509::extensions::Extensions;
+use cryptography_x509::{
+    certificate::Certificate,
+    extensions::{NameConstraints, SequenceOfSubtrees, SubjectAlternativeName},
+    name::GeneralName,
+    oid::{NAME_CONSTRAINTS_OID, SUBJECT_ALTERNATIVE_NAME_OID},
+};
+use ops::CryptoOps;
+use policy::{Policy, PolicyError};
+use trust_store::Store;
+use types::DNSName;
+
+#[derive(Debug, PartialEq, Eq)]
+pub enum ValidationError {
+    Policy(PolicyError),
+}
+
+impl From<PolicyError> for ValidationError {
+    fn from(value: PolicyError) -> Self {
+        ValidationError::Policy(value)
+    }
+}
+
+#[derive(Default)]
+pub struct AccumulatedNameConstraints<'a> {
+    pub permitted: Vec<GeneralName<'a>>,
+    pub excluded: Vec<GeneralName<'a>>,
+}
```

Comment:
I don't understand what value this has compared to jsut passing around a namecosntraints instance

Reply:
Not sure -- CCing @tetsuo-cpp as author here 🙂

Reply:
[9793bb2](https://github.com/pyca/cryptography/pull/8873/commits/9793bb25b6467099f5aaff06005fddb6edafc5dd)

Yeah, good point. I've refactored this to just pass around a `Vec<NameConstraints<'a>>`.